controller/clusters.go

```diff
@@ -92,6 +92,61 @@ func (c *ClustersController) ShowAuthClient(ctx *app.ShowAuthClientClustersConte
 	return ctx.OK(&clusters)
 }
 
+// Show returns the single clusters.
+func (c *ClustersController) Show(ctx *app.ShowClustersContext) error {
+	if !auth.IsSpecificServiceAccount(ctx, auth.OsoProxy, auth.Tenant, auth.JenkinsIdler, auth.JenkinsProxy, auth.Auth) {
```

Comment:
I think we should move all our authorization checks to the service layer. It will be less error prone. Otherwise we can accidentally use not protected service method (which returns sensitive cluster information) from another not well protected controller or service. This information is super sensitive and we should be restrictive as much as possible. 
We do the same in our role management in Auth service for example.

Reply:
I could add tests to make sure that all listed SA are allowed (I checked single SA so far), and make sure any other account is rejected (that's already done). Honestly, I like that idea that the controllers handle the rest of the security checks, after the JWT middleware, of course. This is what we do everywhere else, so not doing it so here would seem confusing. Let's discuss about this ;)

Reply:
@alexeykazakov I wasn't aware that we'd copied the service framework into cluster service, so it's nice to see some consistency between projects. :)  On your point about authorization checks, I agree that we should restrict this in the service layer, the same way we do it in auth service.  Your example of role management is a good one, we have code such as this to ensure the current user has the required privileges to perform the operation:

`err := s.requireViewRolesScope(ctx, currentIdentity, resourceID)`

While the cluster service won't have anything equivalent to auth's permission service, at the abstract level checking for a specific service account is still equivalent to checking a user's scopes; they both perform a security check and therefore we should follow a consistent architecture here.

Reply:
sure! thanks for your feedback ;)

Reply:
after discussion with @alexeykazakov, this change will finally be handled in https://github.com/fabric8-services/fabric8-cluster/issues/53